### PR TITLE
[compiler-rt] prctl interception update, SECCOMP_MODE_FILTER support.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc [hunk 1]
```diff
@@ -1274,6 +1275,8 @@ INTERCEPTOR(int, prctl, int option, unsigned long arg2, unsigned long arg3,
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg5), sizeof(u64));
   } else if (res != -1 && option == PR_GET_PDEATHSIG) {
     COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg2), sizeof(int));
+  } else if (res != -1 && option == PR_SET_SECCOMP) {
+    COMMON_INTERCEPTOR_WRITE_RANGE(ctx, (u64 *)(arg3), struct_sock_fprog_sz);
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:1279]: SECCOMP_MODE_FILTER only?

**Comment by vitalybuka** [comment]
Please don't forget to click "re-request review"

LGTM with clang-format

**Review by vitalybuka — APPROVED** [review]

**Comment by kstoimenov** [comment]
@devnexen, this probably broke the Sanitizer Android build: https://lab.llvm.org/buildbot/#/builders/186/builds/2622. Please take a look.

**Comment by nico** [comment]
Reverted in 99ade15d192db4afa897a7052a9c73dd42c2b88c for now.

**Comment by devnexen** [comment]
Thanks for reporting I ll fix the Android case when I reapply.

**Pull request merged**